data/summary/types/summary.go

```diff
@@ -346,7 +353,7 @@ func removeExcessBuckets[A BucketDataPt[T], T BucketData](buckets *[]*Bucket[A,
 	*buckets = (*buckets)[excess:]
 }
 
-func AddData[A BucketDataPt[T], T BucketData, R RecordTypes, D RecordTypesPt[R]](buckets *[]*Bucket[A, T], userData []D) error {
+func AddData[A BucketDataPt[T], T BucketData, R RecordTypes, D RecordTypesPt[R]](buckets *[]*Bucket[A, T], userData []D, realtimeUpload map[string]bool) error {
```

Comment:
Consider adding a type alias with a `IsRealtimeUpload(uploadId)` method to improve readability.

Reply:
That makes sense, plus allows for handling the missing uploadId case cleanly, which is an assumption the function makes about its input, we can make it sparse with this.